### ISSUE TITLE
[Feature-Request] Allow logseq macros to be rendered in mark map

### ISSUE THREAD

**Issue by Shiv2205** [vipzhicheng/logseq-plugin-mark-map#87]
I have defined a few macros which allow me to color certain words in my notes to draw attention. They work fine in logseq, but mark map doesn't render them.

```edn
:macros 
{
  :colored-text "[:span {:style \"color: $1\"} $2]"
  :yellow "[:span {:style \"color: yellow\"} $1]"
  :yellow-bold "[:span {:style \"color: yellow; font-weight: bold;\"} $1]"
  :red "[:span {:style \"color: red\"} $1]"
  :red-bold "[:span {:style \"color: red; font-weight: bold;\"} $1]"
  :green "<span style=\"color: #00ff4c;\">$1</span>"
  :green-bold "[:span {:style \"color: #00ff4c; font-weight: bold;\"} $1]"
}
```

Example:

- Logseq: 
  <img width="512" alt="Screenshot 2024-10-10 at 10 02 19" src="https://github.com/user-attachments/assets/e3ee8f46-4685-4bff-b8c6-612c2a2275f7">

- MarkMap:
  <img width="389" alt="Screenshot 2024-10-10 at 10 03 27" src="https://github.com/user-attachments/assets/d481e39b-0af3-4619-a44a-a3aec4c8078c">

Here is the raw text: 
```md
<span style="color:red;font-weight:bold;">Only one task </span>should have actions {{red-bold "that write to a shared variable."}}
```

Basic HTML does render (cuz it's ultimately MD), but writing `<span style="color:red;font-weight:bold;">` every time is long. I wanna be able to spend less time annotating my notes, and more time reading/visualizing them with MarkMap.

**Comment by speendo** [comment]
This would indeed be great!

**Comment by vipzhicheng** [comment]
There is no Macro render API in Logseq plugin API. It's hard to parse Macro string especially for the hiccup syntax.
